src/NHibernate/Criterion/QueryOver.cs

```diff
@@ -280,7 +284,8 @@ IQueryOver<TRoot> IQueryOver<TRoot>.ReadOnly()
 	/// Implementation of the <see cref="IQueryOver&lt;TRoot, TSubType&gt;"/> interface
 	/// </summary>
 	[Serializable]
-	public class QueryOver<TRoot,TSubType> : QueryOver<TRoot>, IQueryOver<TRoot,TSubType>
+	public class QueryOver<TRoot,TSubType> : QueryOver<TRoot>, IQueryOver<TRoot,TSubType>,
+		ISupportEntityJoinQueryOver<TRoot>
```

Comment:
I have not seen the reason for implementing this interface on this class rather than on `QueryOver<TRoot>`. Why this choice?

Reply:
Yeah I've just put it next to existing `JoinAlias`/`JoinQueryOver` methods. Looks more natural to have them in the same place.